packages/core/src/utils.ts

```diff
@@ -0,0 +1,17 @@
+type ObjectLiteral = { [key: string]: unknown };
```

Comment:
1. Does this break typings?

```suggestion
type ObjectLiteral = { [key: string | symbol]: unknown };
```

2. Might be nice to document how this differs from the `object` type?